editoast/src/views/rolling_stocks.rs

```diff
@@ -0,0 +1,275 @@
+/// Endpoints
+/// =================================
+
+/// Return a specific set of electrical profiles
+#[get("/{rolling_stock}")]
+async fn get(db_pool: Data<DbPool>, rolling_stock: Path<i64>) -> Result<Json<RollingStock>> {
+    let rolling_stock = rolling_stock.into_inner();
+    block(move || {
+        let mut conn = db_pool.get().expect("Failed to get DB connection");
+        Ok(Json(RollingStock::retrieve(&mut conn, rolling_stock)?))
+    })
+    .await
+    .unwrap()
+}
+
+/// =================================
+/// Queryables
+/// =================================
+
+#[derive(Debug, PartialEq, Queryable, Insertable, Identifiable, Serialize)]
+#[diesel(table_name = osrd_infra_rollingstock)]
+pub struct RollingStock {
+    pub id: i64,
+    pub name: String,
+    pub version: String,
+    pub effort_curves: JsonValue,
+    pub power_class: String,
+    pub length: f64,
+    pub max_speed: f64,
+    pub startup_time: f64,
+    pub startup_acceleration: f64,
+    pub comfort_acceleration: f64,
+    pub gamma: JsonValue,
+    pub inertia_coefficient: f64,
+    pub features: Vec<String>,
+    pub mass: f64,
+    pub rolling_resistance: JsonValue,
+    pub loading_gauge: String,
+    pub metadata: JsonValue,
+}
+
+impl RollingStock {
+    pub fn retrieve(conn: &mut PgConnection, rolling_stock_id: i64) -> Result<RollingStock> {
+        match dsl::osrd_infra_rollingstock
+            .find(rolling_stock_id)
+            .first(conn)
+        {
+            Ok(rolling_stock) => Ok(rolling_stock),
+            Err(DieselError::NotFound) => Err(RollingStockError::NotFound(rolling_stock_id).into()),
+            Err(e) => Err(e.into()),
+        }
+    }
+}
+
+#[derive(Debug, Error, EditoastError)]
+#[editoast_error(base_id = "rollingstocks", context = "Self::context")]
+pub enum RollingStockError {
+    /// Couldn't find the rolling stock with the given id
+    #[error("Rolling stock '{0}', could not be found")]
+    #[editoast_error(status = 404)]
+    NotFound(i64),
+}
+
+impl RollingStockError {
+    fn context(&self) -> Map<String, JsonValue> {
+        match self {
+            RollingStockError::NotFound(rolling_stock_id) => json!({
+                "rolling_stock_id": rolling_stock_id,
+            })
+            .as_object()
+            .cloned()
+            .unwrap(),
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::RollingStock;
+    use crate::client::PostgresConfig;
+    use crate::tables::osrd_infra_rollingstock::dsl;
+    use actix_http::StatusCode;
+    use diesel::prelude::*;
+    use diesel::result::Error;
+    use serde_json::json;
+
+    fn test_rolling_stock_set_transaction(fn_test: fn(&mut PgConnection, RollingStock)) {
+        let mut conn = PgConnection::establish(&PostgresConfig::default().url()).unwrap();
+        conn.test_transaction::<_, Error, _>(|conn| {
+            diesel::delete(dsl::osrd_infra_rollingstock).execute(conn)?;
+
+            let rolling_stock = RollingStock {
```

Comment:
We could have an example of rolling stock in a test directory and load it from the test.